packages/api/src/utils/types.ts

```diff
@@ -178,6 +178,22 @@ export function WithExecutionOptimistic<T extends {data: unknown}>(
   };
 }
 
+/**
+ * SSZ factory helper to wrap an existing type with `{finalized: boolean}`
+ */
+export function WithFinalized<T extends {data: unknown}>(type: TypeJson<T>): TypeJson<T & {finalized: boolean}> {
```

Comment:
`finalized` doesn't appear alone in the endpoints but is always bundled with `executionOptimistic`. 
I think it's better to have `WithExecutionOptimisticAndFinalized()` instead.

Note `executionOptimistic` does appear alone in some endpoints (eg. `getProposerDuties`) so we should retain `WithExecutionOptimistic()`.

Reply:
yeah I know, but you this would require to add `ContainerDataExecutionOptimisticAndFinalized` and `WithExecutionOptimisticAndFinalized`

https://github.com/ChainSafe/lodestar/blob/f2ec0d42365b45495240ce65c27f4c6cdbd2d657/packages/api/src/utils/types.ts#L118

The main issue on unstable branch is that data and metadata is not cleanly separated, on the refactored branch the `ContainerData` wrapper is no longer required and there will be a `ExecutionOptimisticAndFinalized` codec.

All the code in this PR in `@lodestar/api` package has to be rewritten on the other branch as  endpoint definitions are completely different.